pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -1953,15 +1953,46 @@ public AuthorizationService getAuthorizationService() {
         return authorizationService;
     }
 
-    public CompletableFuture<Void> removeTopicFromCache(String topic) {
+    public CompletableFuture<Void> removeTopicFromCache(String topicNameString, Topic topic) {
+        if (topic == null){
+            return removeTopicFromCache(topicNameString);
+        }
+        final CompletableFuture<Optional<Topic>> createTopicFuture = topics.get(topicNameString);
+        // If not exists in cache, do nothing.
+        if (createTopicFuture == null){
+            return CompletableFuture.completedFuture(null);
+        }
+        // We don't need to wait for the future complete, because we already have the topic reference here.
+        if (!createTopicFuture.isDone()){
```

Comment:
we can save us from using getNow and we can chain the CompletableFuture with "thenCompose"

Reply:
> I think that we should wait for this future to complete, otherwise it may create another Topic

One comment suggestion here: "don't need to wait for the future complete" 
https://github.com/apache/pulsar/pull/16247#issuecomment-1170101266

> we can save us from using getNow and we can chain the CompletableFuture with "thenCompose"

Already use "thenCompose" instead "getNow". Thanks